### PR TITLE
Consul Kubernetes Datadog Integration Docs Update

### PR DESCRIPTION
**Pull request by natemollica-nm**
### Description

Adding in an Observability piece to supplement https://github.com/hashicorp/consul-k8s/pull/3407 which added
helm configuration options for configuring Datadog metrics integration with consul-k8s.

### Testing & Reproduction steps

Worked with the information [here](https://github.com/hashicorp/consul/blob/main/website/README.md) to update Consul documentation website with details outlining the changes made in the PR.

### Links

* [Consul Integration (Datadog Checks)](https://docs.datadoghq.com/integrations/consul/?s=unix&tab=containerized)
* [Consul Integration Source Code](https://github.com/DataDog/integrations-core/blob/master/consul)
* [DogstatsD over UDP](https://docs.datadoghq.com/developers/dogstatsd/?tab=kubernetes)
* [DogstatsD over Unix Domain Socket](https://docs.datadoghq.com/developers/dogstatsd/unix_socket/?tab=kubernetes)

### PR Checklist

* [X] external facing docs updated
* [X] appropriate backport labels added
* [X] not a security concern

### CODE REVIEW

File: website/content/docs/k8s/connect/observability/datadog.mdx [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+    The <a href="https://github.com/DataDog/integrations-core/blob/07c04c5e9465ba1f3e0198830896d05923e81283/consul/datadog_checks/consul/consul.py#L55-L61">consul.py</a> integration source code, as well as the <a href="https://github.com/hashicorp/consul-k8s/blob/4cac70496788f50354f96e9331003fcf338f419c/charts/consul/templates/_helpers.tpl#L595-L598">consul-k8s helm chart</a> prohibit the enablement of more that one integration at a time.</em>
+</Highlight>
```
[the remark this answers is not in the document]
Comment (natemollica-nm) [reply website/content/docs/k8s/connect/observability/datadog.mdx]: Updated to markdown table

File: website/content/docs/k8s/connect/observability/datadog.mdx [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+The list of Consul's Prometheus metrics scraped and mapped by this method are listed in the latest [metrics.py](https://github.com/DataDog/integrations-core/blob/master/consul/datadog_checks/consul/metrics.py) of the integration source code.
+
+To understand how Consul Latency metrics are calculated, review the [Consul Network Coordinates](https://developer.hashicorp.com/consul/docs/architecture/coordinates) documentation.
+
+Review the [Datadog Documentation](https://docs.datadoghq.com/integrations/consul/?tab=containerized#data-collected) for the full description of Metrics data collected.
+
+## Openmetrics Prometheus
+
+### Helm Chart Configuration
```
[the remark this answers is not in the document]
Comment (natemollica-nm) [reply website/content/docs/k8s/connect/observability/datadog.mdx]: Done!

File: website/content/docs/k8s/connect/observability/datadog.mdx [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+      </tr>
+      </thead>
+      <tbody>
+      <tr>
+        <td>Does <strong>not</strong> require <a href="https://kubernetes.io/docs/concepts/storage/volumes/#hostpath">hostPath</a> Volume attachment<br/><br/>(<strong><em>KubeDNS</em></strong>) Does <strong>not</strong> require Hostport exposure if accessible from cluster<br/><br/>Similar <code>IP:Port</code> configuration as Virtual Machine hosts</td>
+        <td><strong>No</strong> packet error handling<br/><br/>(<strong><em>Hostport</em></strong>) Requires a networking provider that adheres to the CNI specification, such as Calico, Canal, or Flannel.<br/><br/>(<strong><em>Hostport</em></strong>) Requires port to be exposed on host using <code>hostNetwork</code><br/><br/>(<strong><em>Hostport</em></strong>) Requires firewall access controls to permit access<br/><br/>(<strong><em>Hostport</em></strong>) Network Namespace sharing is required</td>
+        <td>Kubernetes Service <code>IP:Port</code><br/><br/> <strong>or</strong> <br/><br/>Container host port</td>
+      </tr>
+      </tbody>
+    </table>
```
[the remark this answers is not in the document]
Comment (natemollica-nm) [reply website/content/docs/k8s/connect/observability/datadog.mdx]: Done!

File: website/content/docs/k8s/connect/observability/datadog.mdx [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        enabled: true
+  ```
+
+  </CodeBlockConfig>
+
+This method of integration is useful for cases where Prometheus enabled scrapes are desired, and further customization of the collected data is desired.
+
+By default, all metrics pulled using this method will scrape Consul metrics using the `/v1/agent/metrics?format=prometheus` API query, and are considered to be **_custom metrics_**.
+
+Use of this method will map to Datadog as described in [Mapping Prometheus Metrics to Datadog Metrics](https://docs.datadoghq.com/integrations/guide/prometheus-metrics/?tab=latestversion) and summarized below.
```
[the remark this answers is not in the document]
Comment (natemollica-nm) [reply website/content/docs/k8s/connect/observability/datadog.mdx]: updated!

File: website/content/docs/k8s/connect/observability/datadog.mdx [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+|-------------------------|------------------------------------|
+| `Gauge`                 | `gauge`                            |
+| `Counter`               | `count`                            |
+| Histogram: `_count `    | `count.count`                      |
+| Histogram: `_sum`       | `count.sum`                        |
+| Histogram: `_bucket`    | `count.bucket` \|\| `distribution` |
+| Summary: `_count`       | `count.count`                      |
+| Summary: `_sum`         | `count.sum`                        |
+| Summary: `sample`       | `gauge.quantile`                   |
+
```
[the remark this answers is not in the document]
Comment (natemollica-nm) [reply website/content/docs/k8s/connect/observability/datadog.mdx]: Updated!

File: website/content/docs/k8s/connect/observability/datadog.mdx [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+
+#### Metrics Data Collected
+
+  - Collects Consul related metrics via the `/v1/agent/metrics` API endpoint leveraging the `prometheus` query parameter.
+  - Default configuration, uses a wildcard `".*"` to collect all metrics emitted from the endpoint, therefore the list of metrics scraped consists of those listed in the [Agent Telemetry](https://developer.hashicorp.com/consul/docs/agent/telemetry) documentation.
```
[the remark this answers is not in the document]
Comment (natemollica-nm) [reply website/content/docs/k8s/connect/observability/datadog.mdx]: You're right. Don't know why this was as a list. Thank you! Updated!

**Comment by david-yu** [comment]
@natemollica-nm Thank you for doing this and I think a huge boost to what we can do with DataDog. I have a high level question around placement of this doc. It's currently place under Service Mesh -> Observabililty. Are these metrics more for gathering Consul server metrics via DataDog or are these for collecting metrics for all service mesh components?

**Comment by natemollica-nm** [comment]
@david-yu -- Good point to bring up on the placement. This is more Consul specific metrics collection from the servers and nothing regarding Envoy metrics. That's a separate set of annotations that should be placed on Kube pods that have dataplane containers with them. Any recommendations on better placement or where you'd like to see it?

**Comment by david-yu** [comment]
I think moving it under Deployment Configs makes the most sense for me perhaps: https://developer.hashicorp.com/consul/docs/k8s/deployment-configurations/datadog-metrics 

I would put it under Vault Secrets Backend in the nav bar. 

image

File: website/data/docs-nav-data.json [hunk 7]
```diff
@@ -1406,6 +1406,10 @@
                 "path": "k8s/deployment-configurations/vault/wan-federation"
               }
             ]
+          },
+          {
+            "title": "Datadog",
```
Comment (david-yu) [inline website/data/docs-nav-data.json:1411]:
```suggestion
            "title": "Datadog metrics",
```

**Review by david-yu — APPROVED** [review]
Awesome, great work!

**Comment by natemollica-nm** [comment]
@david-yu @boruszak -- Sorry for this last commit and push 😬. I added some additional amplifying information in regard to the Consul Datadog Checks and OpenMetrics sections (Notes) in regard to how Consul RPC TLS certificates are handles by the automated integration. Let me know if these changes are OK with you guys and I'll go ahead and merge.

**Review by david-yu — APPROVED** [review]
Changes LGTM

**Pull request merged**